server/sequencer_manager.go

```diff
@@ -0,0 +1,139 @@
+package server
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian"
+	"github.com/google/trillian/log"
+	"github.com/google/trillian/util"
+)
+
+// SequencerManager controls sequencing activities for logs. At the moment it's very simple
+// with a single task sequencing active logs one at a time. This will be expanded later.
+
+type SequencerManager struct {
+	// done is a channel that provides an exit signal
+	done chan struct{}
+	// storageProvider is the log storage provider used to build sequencers
+	storageProvider LogStorageProviderFunc
+	// batchSize is the batch size to be passed to sequencers run by this manager
+	batchSize int
+	// sleepBetweenLogs is the time to pause after each batch
+	sleepBetweenLogs time.Duration
+	// sleepBetweenRuns is the time to pause after all active logs have processed a batch
+	sleepBetweenRuns time.Duration
+	// runLimit is a limit on the number of sequencing passes. It can only be set for tests
+	runLimit int
+	// timeSource allows us to mock this in tests
+	timeSource util.TimeSource
+}
+
+func NewSequencerManager(done chan struct{}, storageProvider LogStorageProviderFunc, batchSize int, sleepBetweenLogs, sleepBetweenRuns time.Duration) *SequencerManager {
+	return &SequencerManager{done: done, storageProvider: storageProvider, batchSize: batchSize, sleepBetweenLogs: sleepBetweenLogs, sleepBetweenRuns: sleepBetweenRuns, timeSource: new(util.SystemTimeSource)}
+}
+
+// For use by tests, arranges for the sequencer to exit after a number of passes
+func newSequencerManagerForTest(done chan struct{}, storageProvider LogStorageProviderFunc, batchSize int, sleepBetweenLogs, sleepBetweenRuns time.Duration, runLimit int, timeSource util.TimeSource) *SequencerManager {
+	return &SequencerManager{done: done, storageProvider: storageProvider, batchSize: batchSize, sleepBetweenLogs: sleepBetweenLogs, sleepBetweenRuns: sleepBetweenRuns, runLimit: runLimit, timeSource: timeSource}
+}
+
+func (s SequencerManager) sequenceActiveLogs(logIDs []trillian.LogID) bool {
+	// TODO(Martin2112): Demote logging to verbose level
+	glog.Infof("Beginning sequencing run for %d active log(s)", len(logIDs))
+
+	successCount := 0
+	leavesAdded := 0
+
+	for _, logID := range logIDs {
+		// See if it's time to quit
+		select {
+		case <-s.done:
+			return true
+		default:
+		}
+
+		// TODO(Martin2112): Probably want to make the sequencer objects longer lived to
+		// avoid the cost of initializing their state each time but this works for now
+		storage, err := s.storageProvider(logID.TreeID)
+
+		// TODO(Martin2112): Honour the sequencing enabled in log parameters, needs an API change
+		// so deferring it
+		if err == nil {
```

Comment:
indent error flow rather than business-as-usual.
(That might mean either having a separate `time.Sleep` and a `continue`, but that's not necessarily a bad thing - you may want to have a back-off strategy here for failures at some point?)